src/jdk.dynalink/share/classes/jdk/dynalink/BiClassValue.java

```diff
@@ -96,33 +102,47 @@
             }
         }
 
-        private volatile Map<Class<?>, T> forward;
-        private volatile Map<Class<?>, T> reverse;
+        private volatile Map<Class<?>, T> forward = Map.of();
+        private volatile Map<Class<?>, T> reverse = Map.of();
```

Comment:
Since maps are immutable now and allow safe publication via data race (i.e. their internal state is composed of final fields), you could relax the forward/referse fields to be plain fields. You would just have to leave them uninitialized then and check for null at some places if BiClassValues could be published via data race since I don't know whether ClassValue guarantees that the associated values are published safely or not. If it does, then you could pre-initialize the field with empty maps as now, if it does not, then even volatile modifiers don't guarantee that unsafely published BiClassValues object can only be seen with the fields initialized. i.e. you could read null from them.